src/util/list.c

```diff
@@ -118,18 +123,32 @@ int append_list(list_t *list, void *element)
 
 void for_each_list(list_t *list, listIterator iterator)
 {
-    assert(iterator != NULL);
 
     list_node_t *node = list->head;
-    bool result = true;
+    BOOL result = TRUE;
 
-	while(node != NULL && result) {
-		result = iterator(node);
+    assert(iterator != NULL);
+
+    while(node != NULL && result) {
+        result = iterator(node);
+        node = node->next;
+    }
+}
+
+int get_index(list_t *list, int key)
```

Comment:
This function makes no sense. You don't need both. A key can do everything an index can do.